src/tt_scheduler.erl

```diff
@@ -0,0 +1,137 @@
+%%%-------------------------------------------------------------------
+%%% @author Wojciech Pachuta
+%%%-------------------------------------------------------------------
+-module(tt_scheduler).
+-author("Wojciech Pachuta").
+
+-behaviour(gen_server).
+
+%% API
+-export([start_link/0, schedule/3, cancel_schedule/1]).
+
+%% gen_server callbacks
+-export([init/1,
+  handle_call/3,
+  handle_cast/2,
+  handle_info/2,
+  terminate/2,
+  code_change/3]).
+
+%% auxiliary functions
+-export([time_keeper_function/4,  prompt_publishing/1]).
+
+-define(SERVER, ?MODULE).
+
+-record(state, {refs :: #{reference() => pid()}}).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+-spec(start_link() ->
+  {ok, Pid :: pid()} | ignore | {error, Reason :: term()}).
+start_link() ->
+  gen_server:start_link({local, ?SERVER}, ?MODULE, [], []).
+
+-spec(schedule(calendar:datetime(), calendar:datetime(), calendar:time()) -> atom()).
+schedule(StartTime, EndTime, PublishInterval) ->
+  gen_server:call(?SERVER, {schedule, StartTime, EndTime, PublishInterval}).
+
+-spec(cancel_schedule(reference()) -> no_return()).
+cancel_schedule(Ref) ->
+  gen_server:cast(?SERVER, {cancel_schedule, Ref}).
+
+%%%===================================================================
+%%% gen_server callbacks
+%%%===================================================================
+
+-spec init([]) -> {ok, State :: #state{}}.
+init([]) ->
+  process_flag(trap_exit, true),
+  {ok, #state{refs = maps:new()}}.
+
+-spec handle_call({schedule, StartTime :: calendar:datetime(), EndTime :: calendar:datetime(), PublishInterval :: calendar:time()},
+    From :: {pid(), Tag :: term()}, State :: #state{}) ->
+  {reply, reference(), NewState :: #state{}}.
+handle_call({schedule, StartTime, EndTime, PublishInterval}, _From, State) ->
```

Comment:
for example, for this function I can write spec
```erlang
-spec handle_call({schedule, StartTime :: integer(), EndTime :: pid(), PublishInterval :: list()},
    From :: {pid(), Tag :: term()}, State :: #state{}) ->
  {reply, reference(), NewState :: #state{}}.
```
and dializer won't detect that everything in first tuple is wrong. Is dializer just that inefective, or am i doing something wrong?